src/amber/support/message_encryptor.cr

```diff
@@ -0,0 +1,67 @@
+
+require "json"
+require "openssl/cipher"
+
+module Amber::Support
+  class MessageEncryptor
+
+    def initialize(@secret, @cipher = "aes-256-cbc", @digest = :sha1, @sign_secret = nil)
+      @verifier = MessageVerifier.new(@sign_secret || @secret, digest: @digest)
+    end
+
+    # Encrypt and sign a message. We need to sign the message in order to avoid
+    # padding attacks. Reference: http://www.limited-entropy.com/padding-oracle-attacks.
+    def encrypt_and_sign(value : Slice(UInt8)) : String
+      verifier.generate(_encrypt(value))
+    end
+
+    def encrypt_and_sign(value : String) : String
+      encrypt_and_sign(value.to_slice)
+    end
+
+    # Decrypt and verify a message. We need to verify the message in order to
+    # avoid padding attacks. Reference: http://www.limited-entropy.com/padding-oracle-attacks.
+    def decrypt_and_verify(value : String) : Slice(UInt8)
+      _decrypt(verifier.verify(value))
+    end
+
+    private def _encrypt(value)
+      cipher = new_cipher
+      cipher.encrypt
+      cipher.key = @secret
+
+      # Rely on OpenSSL for the initialization vector
+      iv = cipher.random_iv
+
+      encrypted_data = MemoryIO.new
+      encrypted_data.write(cipher.update(value))
+      encrypted_data.write(cipher.final)
+
+      "#{::Base64.strict_encode encrypted_data.to_slice}--#{::Base64.strict_encode iv}"
```

Comment:
Using Base64 takes up a lot more space in our limited cookie space. It's already encrypted so what are we gaining by encoding it to base64?